lib/utils/docker.coffee

```diff
@@ -266,3 +313,73 @@ exports.getDocker = (options) ->
 exports.getImageSize = (docker, image) ->
 	docker.getImage(image).inspectAsync()
 	.get('Size')
+
+hasQemu = ->
+	fs = require('mz/fs')
+
+	getQemuPath()
+	.then(fs.stat)
+	.return(true)
+	.catchReturn(false)
+
+getQemuPath = ->
+	resin = require('resin-sdk-preconfigured')
+	path = require('path')
+	fs = require('mz/fs')
+
+	resin.settings.get('binDirectory')
+	.then (binDir) ->
+		# The directory might not be created already,
+		# if not, create it
+		fs.access(binDir)
+		.catch code: 'ENOENT', ->
+			fs.mkdir(binDir)
+		.then ->
+			path.join(binDir, 'qemu-execve')
+
+platformNeedsQemu = ->
+	os = require('os')
+	os.platform() == 'linux'
+
+installQemu = ->
+	request = require('request')
+	fs = require('fs')
+	zlib = require('zlib')
+
+	qemuVersion = 'v2.5.50-resin-execve'
+
+	getQemuPath()
+	.then (qemuPath) ->
+		new Promise (resolve, reject) ->
+			installStream = fs.createWriteStream(qemuPath)
+			qemuUrl = "https://github.com/resin-io/qemu/releases/download/#{qemuVersion}/qemu-execve.gz"
+			request(qemuUrl)
+			.pipe(zlib.createGunzip())
+			.pipe(installStream)
+			.on('error', reject)
+			.on('finish', resolve)
+
+copyQemu = (context) ->
+	path = require('path')
+	fs = require('mz/fs')
+	# Create a hidden directory in the build context, containing qemu
+	binDir = path.join(context, '.resin')
+	binPath = path.join(binDir, 'qemu-execve')
```

Comment:
`getQemuPath` is the global path of qemu on the system. `binPath` here is the path inside the local build context.